Modules/_zoneinfo.c

```diff
@@ -224,8 +224,14 @@ zoneinfo_new_instance(PyTypeObject *type, PyObject *key)
     self = NULL;
 cleanup:
     if (file_obj != NULL) {
+        PyObject *exc, *val, *tb;
+        PyErr_Fetch(&exc, &val, &tb);
```

Comment:
What is the purpose of this?

Is the issue that there might be a live exception and that `file_obj.close()` may itself raise an exception, so we want to save the original one and chain it to the new one?